pgd_splicer/tests.py

```diff
@@ -338,27 +334,46 @@ def test_import_pdb(self):
         # Choose the first protein in the selection file.
         with open(MonkeyPatch.sitefile('cullpdb_selection.txt'), 'r') as f:
             line = f.readline()
+        code = line.split(' ')[0]
 
         # Install that protein into the database with ProcessPDBTask.
         pdbs = [line]
         from ProcessPDBTask import ProcessPDBTask
         task = ProcessPDBTask()
-        task.work(pdbs)
 
-        # Check that all residues have valid secondary structures.
+        # The protein file is in the local file directory.
+        with MonkeyPatch():
+            task.work(pdbs)
+
+        # Confirm that that protein was added.
         from pgd_core.models import Protein
         try:
-            p = Protein.objects.get(code=line.split(' ')[0])
+            p = Protein.objects.get(code=code)
         except Protein.DoesNotExist:
             self.fail("target protein not in database -- add failed")
         except:
             self.fail("Unknown exception")
 
-        from pgd_constants import SS_CHOICES
+        # Check that no residues are malformed.
+        from pgd_constants import AA_CHOICES, SS_CHOICES
+        valid_aa = [x[0] for x in AA_CHOICES]
         valid_ss = [x[0] for x in SS_CHOICES]
         # '-' is valid and not in SS_CHOICES
         valid_ss.append('-')
         for c in p.chains.all():
             for r in c.residues.all():
+                # Is amino acid valid?
```

Comment:
This section of code confirms that the value being stored for the amino acid is one of the legitimate values.

This protects against the same kind of problem we just had with secondary structure -- if we read the wrong column for amino acid, we'll get an invalid value, and then this test will fail.